core/src/main/java/bisq/core/dao/burningman/BurningManPresentationService.java

```diff
@@ -0,0 +1,322 @@
+        return Math.round(targetAmount) - myAmount;
+    }
+
+    public Set<ReimbursementModel> getReimbursements() {
+        if (!reimbursements.isEmpty()) {
+            return reimbursements;
+        }
+
+        reimbursements.addAll(burnTargetService.getReimbursements(currentChainHeight));
+        return reimbursements;
+    }
+
+    public Optional<Set<String>> findMyGenesisOutputNames() {
+        // Optional.empty is valid case, so we use null to detect if it was set.
+        // As it does not change at new blocks its only set once.
+        //noinspection OptionalAssignedToNull
+        if (myGenesisOutputNames != null) {
+            return myGenesisOutputNames;
+        }
+
+        myGenesisOutputNames = daoStateService.getGenesisTx()
+                .flatMap(tx -> Optional.ofNullable(bsqWalletService.getTransaction(tx.getId()))
+                        .map(genesisTransaction -> genesisTransaction.getOutputs().stream()
+                                .filter(transactionOutput -> transactionOutput.isMine(bsqWalletService.getWallet()))
+                                .map(transactionOutput -> BurningManService.GENESIS_OUTPUT_PREFIX + transactionOutput.getIndex())
+                                .collect(Collectors.toSet())
+                        )
+                );
+        return myGenesisOutputNames;
+    }
+
+    public Set<String> getMyCompensationRequestNames() {
+        // Can be empty, so we compare with null and reset to null at new block
+        if (myCompensationRequestNames != null) {
+            return myCompensationRequestNames;
+        }
+        myCompensationRequestNames = myProposalListService.getList().stream()
+                .filter(proposal -> proposal instanceof CompensationProposal)
+                .map(Proposal::getName)
+                .collect(Collectors.toSet());
+        return myCompensationRequestNames;
+    }
+
+    public Map<String, BurningManCandidate> getBurningManCandidatesByName() {
+        // Cached value is only used for currentChainHeight
+        if (!burningManCandidatesByName.isEmpty()) {
+            return burningManCandidatesByName;
+        }
+
+        burningManCandidatesByName.putAll(burningManService.getBurningManCandidatesByName(currentChainHeight));
+        return burningManCandidatesByName;
+    }
+
+    public LegacyBurningMan getLegacyBurningManForDPT() {
+        if (legacyBurningManDPT.isPresent()) {
+            return legacyBurningManDPT.get();
+        }
+
+        // We do not add the legacy burningman to the list but keep it as class field only to avoid that it
+        // interferes with usage of the burningManCandidatesByName map.
+        // Those are the hashes used by legacy BM for burning BTC received from DPT.
+        Set<String> hashes = Set.of("1701e47e5d8030f444c182b5e243871ebbaeadb5e82f",
+                "1701293c488822f98e70e047012f46f5f1647f37deb7");
```

Comment:
Why are these burningman hashes hard-coded?

Reply:
Those are the hashes in the opReturn at burn txs of the BM to tag different use cases (btc fee, del payout tx to refund agent, del payout tx to trader)